ci_scripts/sync_target_database.py

```diff
@@ -54,18 +53,17 @@ def save_target_database(target_database_text: str, output_file_path: Path) -> N
     output_file_path.write_text(target_database_text)
 
 
-def diff_objects_from_json_str(incoming_text: str, current_text: str) -> List[dict]:
-    """Convert two json strings to objects and take the difference."""
-    current = json.loads(current_text)
-    incoming = json.loads(incoming_text)
-    return [i for i in current + incoming if i not in current or i not in incoming]
+def get_boards_added_or_removed(offline_targets: MbedTargetsOffline, online_targets: MbedTargetsOnline) -> List[dict]:
+    """Check boards added and removed in relation to the offline target database."""
+    added = online_targets - offline_targets
```

Comment:
Am I right in thinking that online db update to any of the properties `__eq__` relies on will trigger a removal and addition for the same board here?
This also reminds me of the fact that we tend to bump "mbed os support" features in bulk when new version of it is released. Food for thought if this should be catered for - it'd probably yield updates to hundreds of targets.

Reply:
I think changing the '__eq__' method to check for mbed os support is outside of the scope of this PR. It may be the right thing to do.